legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/ArtifactsRefreshServiceImpl.java

```diff
@@ -342,6 +332,18 @@ private MetadataEventResponse refreshVersionForProject(ProjectData project, Stri
         });
     }
 
+    private void refreshFunctionalDependenciesOfProjectData(String versionId, List<ProjectVersionDependency> newDependencies, Optional<ProjectData> latestProjectData)
+    {
+        latestProjectData.ifPresent(p ->
+        {
+            p.addVersion(versionId);
```

Comment:
we dont add if it is snaphot... look at the prior implementation

Reply:
if (!versionId.equals(VersionValidator.MASTER_SNAPSHOT) && !this.getVersions().contains(versionId))
        {
            this.versions.add(versionId);
        }

refactored this a little, as we already check that in the addVersion method
so dont require the other check